app/src/main/java/com/weatherxm/util/NumberUtils.kt

```diff
@@ -1,35 +1,61 @@
 package com.weatherxm.util
 
 import android.icu.text.CompactDecimalFormat
-import android.icu.text.NumberFormat
-import com.weatherxm.util.Weather.EMPTY_VALUE
+import com.weatherxm.data.DECIMAL_FORMAT_TOKENS
+import com.weatherxm.ui.common.Contracts.EMPTY_VALUE
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
+import org.koin.core.qualifier.named
+import java.math.BigDecimal
 import java.math.RoundingMode
+import java.text.DecimalFormat
+import java.text.NumberFormat
 
 object NumberUtils : KoinComponent {
     private val compactDecimalFormat: CompactDecimalFormat by inject()
     private val numberFormat: NumberFormat by inject()
+    private val decimalFormat: DecimalFormat by inject(named(DECIMAL_FORMAT_TOKENS))
+
+    private const val DIVISOR_WEI_TO_ETH = "1000000000000000000"
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/350#issuecomment-2388457789>#b046d2</a><span> - 10/02/2024, 12:17 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The DIVISOR_WEI_TO_ETH constant is defined as a String, which may lead to inefficient operations and potential precision loss when used in calculations involving Wei to ETH conversions.
</div>
<div id="fix">
<b>Fix:</b> Replace the String constant with a BigInteger or BigDecimal constant to ensure precise calculations when dealing with Wei to ETH conversions.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -19,2 +19,2 @@ object NumberUtils : KoinComponent {
 -    private const val DIVISOR_WEI_TO_ETH = "1000000000000000000"
 +    private val DIVISOR_WEI_TO_ETH = BigInteger("1000000000000000000")
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged